### PR TITLE
Switch to default barostat interval of 25

### PR DESCRIPTION
**Pull request by badisa**
## Benchmarks

Nvidia A10
Cuda Arch 8.6

Get about 6% speed up of hif2a by changing the interval. (778ns/day vs 826.8ns/day) for an interval that we suspect doesn't impact results.

### Interval of 15 for production

```
hif2a-apo: N=8805 speed: 1281.56ns/day dt: 2.5fs (ran 100000 steps in 16.86s)
hif2a-apo-barostat-interval-25: N=8805 speed: 1066.94ns/day dt: 2.5fs (ran 100000 steps in 20.25s)
hif2a-rbfe-barostat-interval-15: N=8840 speed: 778.00ns/day dt: 2.5fs (ran 100000 steps in 27.77s)
hif2a-rbfe-local: N=8840 speed: 1339.52ns/day dt: 2.5fs (ran 100000 steps in 16.13s)
```

`tests/test_benchmark_free_energy.py::test_benchmark_hif2a_single_topology[complex-5-True]`

```
hrex: True
host: complex
n_windows: 5
n_frames: 100
total ns/day: 616.4
```

### Interval of 25 for Production

```
hif2a-apo: N=8805 speed: 1287.46ns/day dt: 2.5fs (ran 100000 steps in 16.78s)
hif2a-apo-barostat-interval-25: N=8805 speed: 1070.42ns/day dt: 2.5fs (ran 100000 steps in 20.18s)
hif2a-rbfe-barostat-interval-25: N=8840 speed: 826.89ns/day dt: 2.5fs (ran 100000 steps in 26.13s)
hif2a-rbfe-local: N=8840 speed: 1337.66ns/day dt: 2.5fs (ran 100000 steps in 16.15s)
```

`tests/test_benchmark_free_energy.py::test_benchmark_hif2a_single_topology[complex-5-True]`

```
hrex: True
host: complex
n_windows: 5
n_frames: 100
total ns/day: 644.9
```

### Tyk2 at 8a32004c9d0259afe596f8ef3bed8b7ba439605f
tyk2_092823_seed_2023

### Tyk2 at db0fea38438bf886cdf5607a901e2da932c88bfe
* Makes the number of barostat moves per frame identical https://github.com/proteneer/timemachine/commit/db0fea38438bf886cdf5607a901e2da932c88bfe
* Generate Langevin noise in float https://github.com/proteneer/timemachine/commit/8eaa71b81b374b78c23c081e3428be788a256d8b
tyk2-default

### Tyk2 with Barostat interval of 25 during production

tyk2-default

## Todo

- [x] Test against TYK2 to ensure this doesn't impact things.

### CODE REVIEW

File: timemachine/fe/free_energy.py [hunk 1]
```diff
@@ -926,9 +933,16 @@ def log_q_fn(replica_idx: ReplicaIdx, state_idx: StateIdx) -> float:
...
+    # Setup the barostat to run more often for equilibration
+    equil_barostat_interval = 15
+    barostat = context.get_barostat()
+    if barostat is not None:
+        barostat.set_interval(equil_barostat_interval)
+
     def get_equilibrated_xvb(xvb: CoordsVelBox, params: NDArray) -> CoordsVelBox:
         if md_params.n_eq_steps == 0:
             return xvb
+        # Set the movers to 0 to ensure they all equilibrate the same way
```
Comment (mcwitt) [inline timemachine/fe/free_energy.py:945]: nit: was this intentional? (redundant comment)

Comment (badisa) [reply timemachine/fe/free_energy.py:945]: Nope, will remove

File: timemachine/fe/free_energy.py [hunk 2]
```diff
@@ -402,10 +402,17 @@ def sample_with_context(
 ) -> Trajectory:
     # burn-in
     if md_params.n_eq_steps:
+        barostat = ctxt.get_barostat()
+        default_interval = 0 if barostat is None else barostat.get_interval()
+        equil_barostat_interval = 15
```
Comment (mcwitt) [inline timemachine/fe/free_energy.py:407]: Do we think it's really necessary to use different intervals for equilibration and production? Seems simpler to just switch to 25 in both places unless we have evidence that this is insufficient for equilibration.

Comment (mcwitt) [reply timemachine/fe/free_energy.py:407]: nit, if we decide to keep the shorter interval for equilibration: I think a short comment would be useful here, e.g. "set to 15 steps per barostat move for equilibration, then back to the original interval for production"

Comment (badisa) [reply timemachine/fe/free_energy.py:407]: Good question, I would defer to @proteneer on this. 

My intuition is that we want to be 'sure' that the system is at equilibrium before collecting samples and setting a lower interval before collecting samples helps us be certain.

File: timemachine/fe/free_energy.py (hunk 1 above)
Comment (badisa) [reply timemachine/fe/free_energy.py]: Removed in https://github.com/proteneer/timemachine/pull/1221/commits/5942f55f2bbbbb20a8dfc92f862516381e2dd570

File: timemachine/fe/free_energy.py (hunk 2 above)
Comment (badisa) [reply timemachine/fe/free_energy.py:408]: Added comment https://github.com/proteneer/timemachine/pull/1221/commits/abe24775b89b2983c16d0af4f4fa10bb0ba84695, until @proteneer comments on the utility of having two different intervals.

Comment (proteneer) [reply timemachine/fe/free_energy.py:408]: I think the idea was to initially have a higher rate of equilibration when the system may be out  equilibrium (eg. we pad the water box so the density is too low). I don't have a quantitative sense of 15 vs 25 though, I think we'd need to monitor volumes/densities to be more precise here.

**Review by mcwitt — APPROVED** [review]

**Review by proteneer — APPROVED** [review]
Thanks for the plots!

**Pull request merged**